xbmc/settings/AdvancedSettings.cpp

```diff
@@ -1427,29 +1431,62 @@ void CAdvancedSettings::setExtraLogLevel(const std::vector<CVariant> &components
 std::string CAdvancedSettings::GetMusicExtensions() const
 {
   std::string result(m_musicExtensions);
-
-  BinaryAddonBaseList addonInfos;
-  CServiceBroker::GetBinaryAddonManager().GetAddonInfos(addonInfos, true, ADDON_AUDIODECODER);
-  for (const auto& addonInfo : addonInfos)
-  {
-    result += '|';
-    result += CAudioDecoder::GetExtensions(addonInfo);
-  }
+  result += '|' + GetAddonExtensions(ADDON_VFS);
+  result += '|' + GetAddonExtensions(ADDON_AUDIODECODER);
 
   return result;
 }
 
 std::string CAdvancedSettings::GetPictureExtensions() const
 {
   std::string result(m_pictureExtensions);
+  result += '|' + GetAddonExtensions(ADDON_VFS);
+  result += '|' + GetAddonExtensions(ADDON_IMAGEDECODER);
+
+  return result;
+}
+
+std::string CAdvancedSettings::GetVideoExtensions() const
+{
+  std::string result(m_videoExtensions);
+  result += '|' + GetAddonExtensions(ADDON_VFS);
+
+  return result;
+}
+
+std::string CAdvancedSettings::GetSubtitleExtensions() const
+{
+  std::string result(m_subtitlesExtensions);
+  result += '|' + GetAddonExtensions(ADDON_VFS);
 
-  BinaryAddonBaseList addonInfos;
-  CServiceBroker::GetBinaryAddonManager().GetAddonInfos(addonInfos, true, ADDON_IMAGEDECODER);
-  for (auto addonInfo : addonInfos)
+  return result;
+}
+
+void CAdvancedSettings::RegisterAddonExtensions()
+{
+  for (auto const type : ADDON_TYPES)
   {
-    result += '|';
-    result += addonInfo->Type(ADDON_IMAGEDECODER)->GetValue("@extension").asString();
+    std::vector<std::string> extensions;
+    BinaryAddonBaseList addonInfos;
+    CServiceBroker::GetBinaryAddonManager().GetAddonInfos(addonInfos, true, type);
+    for (auto addonInfo : addonInfos)
+    {
+      std::string ext = addonInfo->Type(type)->GetValue(ADDON_VFS ? "@extensions" : "@extension").asString();
```

Comment:
there is by `ADDON_VFS ? ...` a `... == type` missing

Reply:
Maybe possible to change the `@extension` to `@extensions` on the types?

Reply:
Ahh thx will address it asap

Reply:
>Maybe possible to change the @extension to @extensions on the types?

Not sure why this differs for various types. But would be good to unify it.

Reply:
one of these things that happens by accident. sync it up!